File: SmsAuthentication/app/src/main/java/com/example/smsauthentication/MainActivity.java
package com.example.smsauthentication;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.telephony.SmsMessage;
import android.text.InputFilter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    static final int codeLength = 4;

    EditText edtAuthCode;
    Button btnSubmit;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edtAuthCode = findViewById(R.id.edtAuthCode);

        edtAuthCode.setFilters(new InputFilter[] { new InputFilter.LengthFilter(codeLength) });

        btnSubmit = findViewById(R.id.btnSubmit);

        reqPerms();

//        Intent intent = getIntent();
//        processCommand(intent);
    }

    private void reqPerms() {
        String[] permissions = {Manifest.permission.RECEIVE_SMS};
        int permissionCheck = ContextCompat.checkSelfPermission(this, Manifest.permission.RECEIVE_SMS);
        if(permissionCheck == PackageManager.PERMISSION_DENIED){
            ActivityCompat.requestPermissions(this, permissions, 1);
        }
    }
    private void processCommand(Intent intent) {
        if (intent != null) {
            String authCode = intent.getStringExtra("authCode");

            Toast.makeText(getApplicationContext(), "인증번호 확인: " + authCode, Toast.LENGTH_SHORT).show();
            edtAuthCode.setText(authCode);
        }
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        processCommand(intent);
    }
}

File: AndroidKotlin/ImageViewKt/app/src/main/java/com/example/imageviewkt/MainActivity.kt
package com.example.imageviewkt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var i = 1
        bt_toast.setOnClickListener {
            if (i == 1) {
                iv_profile.setImageResource(R.drawable.sunglasssquare)
                i = 0
            } else {
                iv_profile.setImageResource(R.drawable.smilesquare)
                 i = 1
            }
            Toast.makeText(this@MainActivity, "Toast Message Printed.", Toast.LENGTH_SHORT).show()
        }
    }
}